### PR TITLE
Add support for NetBSD

### PR DESCRIPTION
**Pull request by darkgeek**
Hi,

I want to add some code to make `Nomad` build and run on `NetBSD`. 

In this PR, `allocdir/fs_netbsd.go` and `host/netbsd.go`  are used for `NetBSD` only, while the df part in `host/unix.go` is reworked, which might affect other Linux/Unix: instead of leveraging on the frozen `syscall` package, where `Statfs` is not available on `NetBSD`, it might be better to switch to  `gopsutil`, which supports more platforms (currently it doesn't support NetBSD though, I have also sent a PR shirou/gopsutil#1530 to it). 

Can you check this PR to see if it's decent enough to be merged? Thanks.

### CODE REVIEW

**Review by tgross — CHANGES_REQUESTED** [review]
Hi @darkgeek! Thanks for opening this PR. A few comments other than the remarks I've left inline:

* Does this need for `gopsutil` to cut a tag with https://github.com/shirou/gopsutil/pull/1530, and then for Nomad to pick up that dependency bump in the go.mod?
* Is this actually enough to run workloads? What task drivers are you using to verify this?

File: command/agent/host/netbsd.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+//go:build netbsd
+// +build netbsd
```
[the remark this answers is not in the document]
Comment (darkgeek) [reply command/agent/host/netbsd.go]: OK, I have removed `darwin.go`, `linux.go`, `netbsd.go`, and put the `mountedPaths()` function to `unix.go` with `unix` build tag. Then I tried on `Linux` and `NetBSD`, it builds and runs well like before.  Though my tests are quite limited.

**Comment by tgross** [comment]
> Yes, my PR has been merged, and gopsutil already has NetBSD support since the [v3.23.9](https://github.com/shirou/gopsutil/releases/tag/v3.23.9) tag, so you might consider bumping to this version.

Ok, but in which case we should do that in this same PR.

**Comment by darkgeek** [comment]
> > Yes, my PR has been merged, and gopsutil already has NetBSD support since the [v3.23.9](https://github.com/shirou/gopsutil/releases/tag/v3.23.9) tag, so you might consider bumping to this version.
> 
> Ok, but in which case we should do that in this same PR.

I'll try to bump that in my branch this week shortly to see if it works.

**Comment by darkgeek** [comment]
> > Yes, my PR has been merged, and gopsutil already has NetBSD support since the [v3.23.9](https://github.com/shirou/gopsutil/releases/tag/v3.23.9) tag, so you might consider bumping to this version.
> 
> Ok, but in which case we should do that in this same PR.

OK, I did that in the commit https://github.com/hashicorp/nomad/pull/18562/commits/bbc0600a8df28493fa4b94bc8a476a2493f0274a, and have tested on my NetBSD box.

**Comment by tgross** [comment]
@darkgeek you've got some merge conflicts here you'll need to resolve.

**Comment by darkgeek** [comment]
> @darkgeek you've got some merge conflicts here you'll need to resolve.

Resolved by doing rebase on `main`. And I also squash my previous commits to a single one.

**Review by tgross — APPROVED** [review]
LGTM!

I've tested this out on Linux and I've had one of my colleagues who's on macOS give this a spin just to make sure nothing got broken there. I've added a changelog entry as well. Once this is green in CI we'll merge it and it'll go out in the upcoming Nomad 1.7.0 release.

Thanks @darkgeek!

**Comment by tgross** [comment]
The `test-ui/finalize` failure is because of some weird CI issue we have around not exposing the token for Percy to external contributors. That needs to get fixed but isn't a blocker for you here (especially as this isn't a UI PR :grinning: )

**Pull request merged**